### PR TITLE
ENGDESK-33564: Map the internal freeswitch cause code to sip cause

### PR DESCRIPTION
**Pull request by izzytelnyx**
Map the internal freeswitch cause code to sip cause

### CODE REVIEW

**Comment by baloeng** [comment]
Based branch should be development and not master

**Review by dev-ryanc — COMMENTED** [review]
Follow the convention of creating the PR, please fix the PR request title

File: src/mod/endpoints/mod_sofia/mod_sofia.c [hunk 1]
```diff
@@ -539,10 +539,10 @@ switch_status_t sofia_on_hangup(switch_core_session_t *session)
 				if ((switch_channel_test_flag(channel, CF_INTERCEPT) || cause == SWITCH_CAUSE_PICKED_OFF || cause == SWITCH_CAUSE_LOSE_RACE)
 					&& !switch_true(switch_channel_get_variable(channel, "ignore_completed_elsewhere"))) {
 					reason = switch_core_session_strdup(session, "SIP;cause=200;text=\"Call completed elsewhere\"");
-				} else if (cause > 0 && cause < 128) {
+				} else if ((cause > 0 && cause < 128) || switch_telnyx_is_custom_q850_code(cause)) {
```
Comment (dev-ryanc) [inline src/mod/endpoints/mod_sofia/mod_sofia.c:542]: Those custom Q850 errors are both from FS and Telnyx. Ideally this function to check for customq850 can be a part of switch_channel instead.

**Review by dev-ryanc — CHANGES_REQUESTED** [review]

File: src/switch_channel.c [hunk 2]
```diff
@@ -417,6 +417,46 @@ SWITCH_DECLARE(switch_call_cause_t) switch_channel_get_cause_q850(switch_channel
 	return switch_channel_cause_q850(channel->hangup_cause);
 }
 
+SWITCH_DECLARE(switch_bool_t) switch_channel_is_custom_q850_code(switch_call_cause_t cause)
+{
+	switch(cause) {
+		case SWITCH_CAUSE_SUCCESS:                  break;
```
Comment (dev-ryanc) [inline src/switch_channel.c:423]: Shift one tab back, case should be the same line with the switch statement

File: src/mod/endpoints/mod_sofia/mod_sofia.c [hunk 3]
```diff
@@ -539,10 +539,10 @@ switch_status_t sofia_on_hangup(switch_core_session_t *session)
 				if ((switch_channel_test_flag(channel, CF_INTERCEPT) || cause == SWITCH_CAUSE_PICKED_OFF || cause == SWITCH_CAUSE_LOSE_RACE)
 					&& !switch_true(switch_channel_get_variable(channel, "ignore_completed_elsewhere"))) {
 					reason = switch_core_session_strdup(session, "SIP;cause=200;text=\"Call completed elsewhere\"");
-				} else if (cause > 0 && cause < 128) {
+				} else if ((cause > 0 && cause < 128) || switch_channel_is_custom_q850_code(cause) == SWITCH_TRUE) {
```
Comment (dev-ryanc) [inline src/mod/endpoints/mod_sofia/mod_sofia.c:542]: No need to explicitly compare to `SWITCH_TRUE`.

**Review by dev-ryanc — APPROVED** [review]
lgtm

**Review by minhtuan1407-telnyx — APPROVED** [review]

**Pull request merged**